codec/src/main/java/io/netty/handler/codec/DelimitedLengthFieldBasedFrameDecoder.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+
+import java.nio.charset.Charset;
+
+/**
+ * Frames input by looking for a length field that is delimited, as opposed to being a fixed length
+ * <p>
+ * Uses the same approach as the {@link DelimiterBasedFrameDecoder} to find delimiter in input
+ */
+public class DelimitedLengthFieldBasedFrameDecoder extends BaseLengthBasedFrameDecoder {
+
+    private final int lengthAdjustment;
+    private final Charset lengthFieldCharset;
+    private final ByteBuf delimiter;
+    private final boolean trimLengthString;
+
+    private boolean discardingTooLongFrame;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) "discardingTooLongFrame" is the name of a field in "BaseLengthBasedFrameDecoder". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2387)